common/streams/src/stream_limit.rs

```diff
@@ -14,38 +14,64 @@ use crate::SendableDataBlockStream;
 
 pub struct LimitStream {
     input: SendableDataBlockStream,
-    limit: usize,
+    limit: Option<usize>,
+    offset: usize,
     current: usize,
 }
 
 impl LimitStream {
-    pub fn try_create(input: SendableDataBlockStream, limit: usize) -> Result<Self> {
+    pub fn try_create(
+        input: SendableDataBlockStream,
+        limit: Option<usize>,
+        offset: usize,
+    ) -> Result<Self> {
         Ok(LimitStream {
             input,
             limit,
+            offset,
             current: 0,
         })
     }
 
     pub fn limit(&mut self, block: &DataBlock) -> Result<Option<DataBlock>> {
         let rows = block.num_rows();
-        if self.current == self.limit {
+
+        let limit = self.limit.unwrap_or(usize::MAX - self.offset);
+        let offset = self.offset;
+        let current = self.current;
+
+        // There are two intervals:
+        // r1: [current, current + rows)
+        // r2: [offset, offset + limit), note that limit may be infinite.
+        //
+        // There are 6 possible relationships between r1 and r2:
+        if current + rows <= offset {
```

Comment:
It's ok, but this code complicated for me. maybe we implement it as `IgnoreHeadStream { ignore_rows: usize }` is better?

for example:
```
SELECT * FROM `table` LIMIT 5, 10; -> LimitStream {limit: 10, input: IgnoreHeadStream {ignore_rows: 5}}
```

Reply:
Good idea! And we can rename them to TakeStream and SkipStream.

Reply:
I will implement them later.

Reply:
The refactor is done